scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts

```diff
@@ -0,0 +1,323 @@
+            {
+                nodeLabel: 'Juliet',
+                nodeValue: '',
+                childNodes: [
+                    {
+                        nodeLabel: 'name',
+                        nodeValue: 'Romeo\'s love',
+                    },
+                ],
+            },
+        ],
+    };
+
+    const DATA_EMPTY_VALUE_OPTIMIZED = {
+        childNodes: [
+            {
+                nodeLabel: 'Juliet',
+                nodeValue: 'Romeo\'s love',
+                childNodes: [],
+                nodeObjectName: 'Romeo\'s love',
+            },
+        ],
+    };
+
+    const DATA_TYPE = {
+        childNodes: [
+            {
+                nodeLabel: 'type',
+                nodeValue: 'MyObjectType',
+            },
+            {
+                nodeLabel: 'name',
+                nodeValue: 'MyObjectName',
+            },
+        ],
+    };
+
+    const DATA_TYPE_OPTIMIZED = {
+        nodeLabel: 'MyObjectType',
+        nodeValue: 'MyObjectName',
+        childNodes: [],
+        nodeObjectType: 'MyObjectType',
+        nodeObjectName: 'MyObjectName',
+    };
+
+    const CHILDREN = {
+        nodeLabel: 'something',
+        nodeValue: 'special',
+        childNodes: [
+            {
+                nodeLabel: 'children',
+                childNodes: [
+                    {
+                        nodeLabel: 'someLabel',
+                    },
+                ],
+            },
+            {
+                nodeLabel: 'otherNode',
+            },
+        ],
+    };
+
+    const CHILDREN_OPTIMIZED = {
+        nodeLabel: 'something',
+        nodeValue: 'special',
+        childNodes: [
+            {
+                nodeLabel: 'otherNode',
+            },
+            {
+                nodeLabel: 'children',
+                childNodes: [
+                    {
+                        nodeLabel: 'someLabel',
+                    },
+                ],
+            },
+        ],
+    };
+
+    const EMPTY_LABEL = {
+        nodeLabel: 'something',
+        nodeValue: 'special',
+        childNodes: [
+            {
+                nodeLabel: '',
+            },
+        ],
+    };
+
+    const EMPTY_LABEL_OPTIMIZED = {
+        nodeLabel: 'something',
+        nodeValue: 'special',
+        childNodes: [
+            {
+                nodeLabel: 'Item',
+            },
+        ],
+    };
+
+    const ITEM_LABEL_ROOT_NODE = {
+        nodeLabel: 'Item',
+    };
+
+    const ITEM_LABEL_ROOT_NODE_OPTIMIZED = {};
+
+    const ITEM_LABEL_WITH_VALUE_ROOT_NODE = {
+        nodeLabel: 'Item',
+        nodeValue: 'someValue',
+    };
+
+    const ITEM_LABEL_WITH_VALUE_ROOT_NODE_OPTIMIZED = {
+        nodeLabel: 'Item',
+        nodeValue: 'someValue',
+    };
+
+    // Arrange: set up new instance of the pipe TreeDataOptimizer
+    beforeEach(() => {
+        scTreeDataOptimizer = new TreeDataOptimizerPipe();
+    });
+
+    // TODO Check with Rolf whether we need to remove empty child nodes
```

Comment:
please check with @bruderol  :-)

Reply:
I think this case is irrelevant - if a user writes weird stuff into its data tress, its going to be displayed weired - we do not have to care about this case - so let's remove it.

Reply:
As resolved together with you